dashboard/app/helpers/users_helper.rb

```diff
@@ -203,30 +235,40 @@ def script_progress_for_users(users, script)
     script:,
     user:,
     user_levels_by_level:,
+    teacher_feedback_by_level:,
     paired_user_levels:,
     include_timestamp: false
   )
     progress = {}
     script.script_levels.each do |sl|
       sl.level_ids.each do |level_id|
         level = Level.cache_find(level_id)
+        level_for_progress = level.get_level_for_progress(user, script)
+        ul = user_levels_by_level.try(:[], level_for_progress.id)
+
+        # For contained levels, progress is stored with the contained level id but feedback is stored with
+        # the level id, which is why we need a different level id to get feedback for contained levels.
+        level_id_for_feedback = level.contained_levels.any? ? level.id : level_for_progress.id
```

Comment:
could we avoid an extra query for non-contained levels if we move line 246 into a `if level.contained_levels.any?` block?

Reply:
`get_level_for_progress` will get the contained level OR the sublevel, it's not just for contained levels unfortunately.